vnext/ReactUWP/Views/Image/ReactImage.cpp

```diff
@@ -65,51 +108,111 @@ winrt::fire_and_forget ReactImage::Source(ImageSource source) {
 
   winrt::Uri uri{Utf8ToUtf16(uriString)};
   winrt::hstring scheme{uri.SchemeName()};
-  bool needsDownload = (scheme == L"http") || (scheme == L"https");
+  bool needsDownload =
+      ((scheme == L"http") || (scheme == L"https")) && (m_useCompositionBrush || !source.headers.empty());
   bool inlineData = scheme == L"data";
+
   // get weak reference before any co_await calls
   auto weak_this{get_weak()};
 
   try {
     m_imageSource = source;
 
-    winrt::InMemoryRandomAccessStream memoryStream;
     if (needsDownload) {
```

Comment:
> if (needsDownload) { [](start = 3, length = 21)

I don't see anything clearing m_memoryStream.  Consider what happens if the Source is first set to a stream, then set to a URI.  when you get to the checks on line 127 you'll have m_memoryStream still set to the old stream, which seems incorrect.

I think you should be nulling out m_memoryStream before doing the co_awaits.